indexer/src/executor.rs

```diff
@@ -90,29 +189,31 @@ impl IndexExecutor {
             handlers.push(handler.handler);
         }
 
-        Ok(IndexExecutor {
+        Ok(WasmIndexExecutor {
             instance,
             _module: module,
             _store: store,
             events,
             db: env.db.clone(),
         })
     }
+}
 
+impl Executor for WasmIndexExecutor {
     /// Restore index from wasm file
-    pub fn from_file(_index: &Path) -> IndexerResult<IndexExecutor> {
+    fn from_file(_index: &Path) -> IndexerResult<Self> {
         unimplemented!()
     }
 
     /// Trigger a WASM event handler, passing in a serialized event struct.
-    pub fn trigger_event(&self, event_name: &str, bytes: Vec<Vec<u8>>) -> IndexerResult<()> {
+    fn trigger_event(&self, event_name: ReceiptEvent, bytes: Vec<Vec<u8>>) -> IndexerResult<()> {
```

Comment:
Changed the input type for `trigger_event` to be a `ReceiptEvent`
- The idea is that a `ReceiptEvent` is just a type, whereas `fuel_tx::Receipt` is a value
- Fuel contracts output values (`fuel_tx::Receipt`) and we trigger execution based on the type of `ReceiptType`
  - We can obviously add filters later to narrow down what type of specific `Receipt` we want to execute on

Reply:
I'd say we should ship the whole list of receipts to the module that's interested in events from a contract, and then have the all boilerplate of receipt handling happen in the modules. The service side would then just maintain a generic subscription list for <contract type> -> <indexer module>  and/or <contract address> -> <indexer module> mappings. We keep the service code pretty general that way and the modules can get more specific. So trigger_event should maybe become private, and then we'd have a dispatch loop in the executor.